code/modules/hydroponics/spreading/spreading_growth.dm

```diff
@@ -170,6 +147,62 @@
 			chem_regen_cooldown = FALSE
 
 
+/obj/effect/plant/proc/life()
+	var/turf/simulated/T = get_turf(src)
+	if(istype(T))
+		health -= seed.handle_environment(T,T.return_air(),null,1)
+	if(health < max_health)
+		//Plants can grow through closed airlocks, but more slowly, since they have to force metal to make space
+		var/obj/machinery/door/D = (locate(/obj/machinery/door) in loc)
```

Comment:
Here's where i previously coded spreading through airlocks. You should remove this code and hook in your special new function instead